packages/rum-recorder/src/domain/rrweb/mutationObserver.spec.ts

```diff
@@ -0,0 +1,635 @@
+
+    it('does not emit a mutation when a text node keeps the same value', () => {
+      snapshot(document)
+      const { controller, callbackSpy } = start()
+
+      textNode.data = 'bar'
+      textNode.data = 'foo'
+      controller.flush()
+
+      expect(callbackSpy).not.toHaveBeenCalled()
+    })
+  })
+
+  describe('attributes mutations', () => {
+    it('emits a mutation when an attribute is changed', () => {
+      const serializedDocument = snapshot(document)[0]!
+      const { controller, callbackSpy, getLatestMutationPayload } = start()
+
+      sandbox.setAttribute('foo', 'bar')
+      controller.flush()
+
+      expect(callbackSpy).toHaveBeenCalledTimes(1)
+
+      const { validate, expectInitialNode } = createMutationPayloadValidator(serializedDocument)
+      validate(getLatestMutationPayload(), {
+        attributes: [
+          {
+            node: expectInitialNode({ idAttribute: 'sandbox' }),
+            attributes: { foo: 'bar' },
+          },
+        ],
+      })
+    })
+
+    it('does not emit a mutation when an attribute keeps the same value', () => {
+      sandbox.setAttribute('foo', 'bar')
+      snapshot(document)
+      const { controller, callbackSpy } = start()
+
+      sandbox.setAttribute('foo', 'biz')
+      sandbox.setAttribute('foo', 'bar')
+      controller.flush()
+
+      expect(callbackSpy).not.toHaveBeenCalled()
+    })
+
+    it('reuse the same mutation when multiple attributes are changed', () => {
+      const serializedDocument = snapshot(document)[0]!
+      const { controller, getLatestMutationPayload } = start()
+
+      sandbox.setAttribute('foo1', 'biz')
+      sandbox.setAttribute('foo2', 'bar')
+      controller.flush()
+
+      const { validate, expectInitialNode } = createMutationPayloadValidator(serializedDocument)
+      validate(getLatestMutationPayload(), {
+        attributes: [
+          {
+            node: expectInitialNode({ idAttribute: 'sandbox' }),
+            attributes: { foo1: 'biz', foo2: 'bar' },
+          },
+        ],
+      })
+    })
+  })
+
+  describe('ignored nodes', () => {
+    let ignoredElement: HTMLElement
+
+    beforeEach(() => {
+      ignoredElement = document.createElement('script')
+      sandbox.appendChild(ignoredElement)
+    })
+
+    it('skips ignored nodes when looking for the next id', () => {
+      const serializedDocument = snapshot(document)[0]!
+
+      const { controller, getLatestMutationPayload } = start()
+
+      sandbox.insertBefore(document.createElement('a'), ignoredElement)
+
+      controller.flush()
+
+      const { validate, expectInitialNode, expectNewNode } = createMutationPayloadValidator(serializedDocument)
+      validate(getLatestMutationPayload(), {
+        adds: [
+          {
+            parent: expectInitialNode({ idAttribute: 'sandbox' }),
+            node: expectNewNode({ type: NodeType.Element, tagName: 'a' }),
+          },
+        ],
+      })
+    })
+
+    describe('does not emit mutations', () => {
```

Comment:
```suggestion
    describe('does not emit mutations occurring in ignored node', () => {
```
to be clearer and consistent with the approach for hidden node